apps/condo/domains/billing/utils/receiptQRCodeUtils.spec.js

```diff
@@ -33,4 +72,174 @@ describe('receiptQRCodeUtils', () => {
+                const resolvers = {
+                    onNoReceipt: jest.fn(),
+                    onReceiptPeriodEqualsQrCodePeriod: jest.fn(),
+                    onReceiptPeriodNewerThanQrCodePeriod: jest.fn(),
+                    onReceiptPeriodOlderThanQrCodePeriod: jest.fn(),
+                }
+                await compareQRCodeWithLastReceipt(qrCodeObj, resolvers)
+
+                expect(resolvers.onNoReceipt).toBeCalledTimes(0)
+                expect(resolvers.onReceiptPeriodEqualsQrCodePeriod).toBeCalledTimes(1)
+                expect(resolvers.onReceiptPeriodNewerThanQrCodePeriod).toBeCalledTimes(0)
+                expect(resolvers.onReceiptPeriodOlderThanQrCodePeriod).toBeCalledTimes(0)
+
+                expect(resolvers.onReceiptPeriodEqualsQrCodePeriod).toHaveBeenCalledWith(billingReceiptForComparison)
+            })
+
+            test('last billing receipt have newer period than qr-code period', async () => {
+                const resolvers = {
+                    onNoReceipt: jest.fn(),
+                    onReceiptPeriodEqualsQrCodePeriod: jest.fn(),
+                    onReceiptPeriodNewerThanQrCodePeriod: jest.fn(),
+                    onReceiptPeriodOlderThanQrCodePeriod: jest.fn(),
+                }
+                await compareQRCodeWithLastReceipt({ ...qrCodeObj, PaymPeriod: '05.2024' }, resolvers)
+
+                expect(resolvers.onNoReceipt).toBeCalledTimes(0)
+                expect(resolvers.onReceiptPeriodEqualsQrCodePeriod).toBeCalledTimes(0)
+                expect(resolvers.onReceiptPeriodNewerThanQrCodePeriod).toBeCalledTimes(1)
+                expect(resolvers.onReceiptPeriodOlderThanQrCodePeriod).toBeCalledTimes(0)
+
+                expect(resolvers.onReceiptPeriodNewerThanQrCodePeriod).toHaveBeenCalledWith(billingReceiptForComparison)
+            })
+
+            test('last billing receipt period is older that qr-code period', async () => {
+                const resolvers = {
+                    onNoReceipt: jest.fn(),
+                    onReceiptPeriodEqualsQrCodePeriod: jest.fn(),
+                    onReceiptPeriodNewerThanQrCodePeriod: jest.fn(),
+                    onReceiptPeriodOlderThanQrCodePeriod: jest.fn(),
+                }
+                await compareQRCodeWithLastReceipt({ ...qrCodeObj, PaymPeriod: '07.2024' }, resolvers)
+
+                expect(resolvers.onNoReceipt).toBeCalledTimes(0)
+                expect(resolvers.onReceiptPeriodEqualsQrCodePeriod).toBeCalledTimes(0)
+                expect(resolvers.onReceiptPeriodNewerThanQrCodePeriod).toBeCalledTimes(0)
+                expect(resolvers.onReceiptPeriodOlderThanQrCodePeriod).toBeCalledTimes(1)
+
+                expect(resolvers.onReceiptPeriodOlderThanQrCodePeriod).toHaveBeenCalledWith(billingReceiptForComparison)
+            })
+        })
+    })
+
+    test('Auxiliary data has correct format', async () => {
+        const [o10n] = await createTestOrganization(adminClient)
```

Comment:
How about test a for the address error case?

Reply:
Added